### PR TITLE
Align errors

### PR DESCRIPTION
**Pull request by freaz**
## Description

Take it as suggestion, I am not still 100% sure if I broke something, so I wanted to open PR and read though it 

I am trying to change it so all errors inherit from native `Error`. 

## Motivation and Context

Welp, I believe error should be an actual error.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](https://github.com/superfaceai/one-sdk-js/blob/main/SECURITY.md) is correct.
- [ ] I have read the [CONTRIBUTING](https://github.com/superfaceai/one-sdk-js/blob/main/CONTRIBUTING.md) document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/core/errors/errors.test.ts [hunk 1]
```diff
@@ -1,6 +1,26 @@
-import { SDKExecutionError, UnexpectedError } from './errors';
+import { ErrorBase, SDKExecutionError, UnexpectedError } from './errors';
 
 describe('errors', () => {
+  describe('ErrorBase', () => {
+    const error = new ErrorBase('MyKind', 'My message');
+
+    it('has has kind', () => {
```
Comment (Jakub-Vacek) [inline src/core/errors/errors.test.ts:7]: has has has :D

File: src/core/errors/errors.test.ts [hunk 2]
```diff
@@ -56,7 +76,7 @@ Hint: hint3
       );
 
       expect(Object.prototype.toString.call(error)).toBe(
-        '[object SdkExecutionError]'
+        '[object SDKExecutionError]'
```
[the remark this answers is not in the document]
Comment (freaz) [reply src/core/errors/errors.test.ts:79]: 🫠 ouch, I didn’t read it after myself yet.

File: src/core/errors/errors.test.ts (hunk 1 above)
Comment (freaz) [reply src/core/errors/errors.test.ts:7]: 🫠 ouch, I didn’t read it after myself yet.

**Comment by freaz** [comment]
Changes in PR were superseded by changed error structure in #258

**Pull request closed without merging**